addons/account/models/account_move_line.py

```diff
@@ -1547,13 +1550,22 @@ def changed(fname):
                     line.balance = amount_currency
 
         after = existing()
+        changed_currency_rate = any(changed_existing_val('currency_rate') for line in after)
+        term_line = False
+        aggregated_balance = 0
         for line in after:
+            if line.display_type == 'payment_term' and changed_currency_rate and len(after) > 1:
+                term_line = line
+                continue
             if (
                 (changed('amount_currency') or changed('currency_rate') or changed('move_type'))
                 and (not changed('balance') or (line not in before and not line.balance))
             ):
                 balance = line.company_id.currency_id.round(line.amount_currency / line.currency_rate)
                 line.balance = balance
+            aggregated_balance += abs(line.balance)
+        if term_line:
+            term_line.balance = aggregated_balance if term_line.amount_currency > 0 else -1 * aggregated_balance
```

Comment:
Not sure I understand what you're doing here. Can you elaborate please?

Reply:
Basically, the non `payment_term` lines will (at least in the tests) not always have the correct sign on their balance and if the raw values are added it will not aggregate correctly. To work around this I am aggregating the absolute value of the line balances and then changing the sign of the payment_term line balance to be the correct sign relative to the amount_currency on the term_line